google/cloud/aiplatform/experimental/vertex_model/base.py

```diff
@@ -160,10 +147,22 @@ def f(*args, **kwargs):
             valid_types = [int, float, str] + list(
                 obj._data_serialization_mapping.keys()
             )
+
             if parameter_type not in valid_types:
-                raise RuntimeError(
-                    f"{parameter_type} not supported. parameter_name = {parameter_name}. The only supported types are {valid_types}"
-                )
+                if parameter_type.__name__ == "DataLoader":
+                    obj._data_serialization_mapping[parameter_type] = (
+                        pytorch._deserialize_dataloader,
+                        pytorch._serialize_dataloader,
+                    )
+                elif parameter_type.__name__ == "DataFrame":
+                    obj._data_serialization_mapping[parameter_type] = (
+                        pandas._deserialize_dataframe,
+                        pandas._serialize_dataframe,
+                    )
+                else:
+                    raise RuntimeError(
+                        f"{parameter_type} not supported. parameter_name = {parameter_name}. The only supported types are {valid_types}"
+                    )
 
```

Comment:
It seems like `_make_source` also needs to be updated to remove the references to serialization modules and perform the same safe deserializer lookup at runtime.